### DISCUSSION TITLE
Gltfpack compression removes uv coordinates

### DISCUSSION THREAD

**Discussion by Junisek** [zeux/meshoptimizer#631] [Q&A] [answered]
Hello, I'm not sure if this is a bug or feature or maybe I'm missing something but my blender exported glb compressed with gltfpack is missing UV coordinates so I'm unable to assign textures in a three js appliacation. I already asked on the three js forum and they suggested I ask here.

The problem only appears with the compressed file, everyting works fine with just the blender export. When I inspect the model in three, uv coordinates under geometry-attributes are missing. I tried to play around with different parameters in the gltfpack command but with no success. (BTW I also noticed some mesh children are missing and also renamed as mesh_0, mesh_1 etc. unless I use the parameters -cc -kn -ke -vpf. This is not an issue anymore as I figured it out, just saying/asking, if this is a desired behavior.)

I would be grateful for any suggestions.

**Comment by zeux** [comment, marked as answer]
Yes, that's expected; the texture coordinates are transformed and the reverse transform needs to be stored in glTF data attached to the texture reference in the material. You can work around this by attaching a dummy texture. See https://github.com/zeux/meshoptimizer/issues/543#issuecomment-1465026962.

To preserve node names, you should just need `-kn`. You may need `-vpf` although I would recommend adjusting the application code such that it doesn't matter whether `-vpf` is used or not.

**Reply by Junisek** [reply]
I see. I tried to solve it with a dummy texture but I didn't know I have to do this `map.repeat.copy(modelMaterial.map.repeat);
map.offset.copy(modelMaterial.map.offset);`
Now it works like a charm. Thank you!
I believe this info would be useful to other people who don't have such a deep understanding of the technology, maybe state it somewhere on the main overview?

**Reply by zeux** [reply]
Yeah I agree, this comes up enough that it warrants a readme update. I'll take a look into this.
